pkg/pipeline/extract/aggregate/aggregate.go

```diff
@@ -203,6 +203,52 @@ func (aggregate Aggregate) Evaluate(entries []config.GenericMap) error {
 	return nil
 }
 
+func (aggregate Aggregate) computeTopK(inputMetrics []config.GenericMap) []config.GenericMap {
+	nItemsInList := int(0)
+	topk := aggregate.Definition.TopK
+
+	outputMetrics := make([]config.GenericMap, topk)
+	minRecentCount := inputMetrics[0]["recent_count"].(int)
+	indexMin := 0
+	log.Debugf("minRecentCount = %d", minRecentCount)
+
+	for index, metric := range inputMetrics {
+		// fill output with first topk items
+		if nItemsInList < topk {
+			outputMetrics[nItemsInList] = metric
+			nItemsInList++
+			count := metric["recent_count"].(int)
+			log.Debugf("item added, count = %d, index = %d", count, index)
+			if count < minRecentCount {
+				minRecentCount = count
+				indexMin = index
+				log.Debugf("minRecentCount = %d, index = %d", minRecentCount, index)
+			}
+			continue
+		}
+
+		currentRecentCount := metric["recent_count"].(int)
+		if currentRecentCount > minRecentCount {
+			// replace item that has minRecentCount
+			outputMetrics[indexMin] = metric
+			log.Debugf("item added, count = %d", currentRecentCount)
+			log.Debugf("replaced index = %d, count = %d", indexMin, minRecentCount)
+			// find updated minRecentCount
+			newMinRecentCount := currentRecentCount
+			for i, m := range outputMetrics {
+				count := m["recent_count"].(int)
+				if count < newMinRecentCount {
+					newMinRecentCount = count
+					indexMin = i
+				}
+			}
+			minRecentCount = newMinRecentCount
```

Comment:
Finding the new min could be faster using a min heap. 
Alternatively, we can use Selection algorithm to find the k-th value in O(n) and then find all the values that are larger.
Do you think it worth the extra work?
https://stackoverflow.com/a/4961130/2749989

Reply:
Assuming the topk value is small, I don't think it is necessary to overly complicate the algorithm. We can discuss further.

Reply:
I changed the code to use a heap of size topk.